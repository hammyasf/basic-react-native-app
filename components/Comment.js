import { Image, StyleSheet, Text, View } from "react-native";

export default function Comment({ comment }) {
  return (
    <View style={styles.container}>
      <View style={styles.innerContainer}>
        <Image
          style={styles.image}
          source={{
            uri: `https://api.dicebear.com/7.x/micah/png?seed=${comment.id}`,
          }}
        />
        <View>
          <Text numberOfLines={1} ellipsizeMode="tail" style={styles.name}>
            {comment.name}
          </Text>
          <Text style={styles.email}>{comment.email}</Text>
        </View>
      </View>

      <Text style={styles.body}>{comment.body}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#333",
    borderRadius: 10,
    padding: 20,
    marginBottom: 20,
  },
  innerContainer: {
    display: "flex",
    flexDirection: "row",
    gap: 20,
    alignContent: "center",
    justifyContent: "flex-start",
  },
  image: {
    width: 40,
    height: 40,
    marginBottom: 20,
  },
  name: {
    fontSize: 20,
    fontWeight: "bold",
    color: "#FFF",
    width: "80%",
  },
  email: {
    fontSize: 14,
    color: "#eef",
    marginBottom: 10,
    marginTop: 5,
  },
  body: {
    fontSize: 16,
    color: "#FFF",
  },
});
