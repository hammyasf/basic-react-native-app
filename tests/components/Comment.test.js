import React from "react";
import renderer from "react-test-renderer";

import Comment from "../../components/Comment";

describe("<Comment />", () => {
  it("has 2 children", () => {
    const tree = renderer
      .create(
        <Comment
          comment={{
            id: 1,
            name: "name",
            email: "email",
            body: "body",
          }}
        />
      )
      .toJSON();
    expect(tree.children.length).toBe(2);
  });
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Comment
          comment={{
            id: 1,
            name: "name",
            email: "email",
            body: "body",
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("has the correct name", () => {
    const tree = renderer
      .create(
        <Comment
          comment={{
            id: 1,
            name: "name",
            email: "email",
            body: "body",
          }}
        />
      )
      .toJSON();
    expect(tree.children[0].children[1].children[0].children[0]).toBe("name");
  });
});
