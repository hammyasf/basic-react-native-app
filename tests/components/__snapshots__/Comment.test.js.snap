// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#333",
      "borderRadius": 10,
      "marginBottom": 20,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "alignContent": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 20,
        "justifyContent": "flex-start",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://api.dicebear.com/7.x/micah/png?seed=1",
        }
      }
      style={
        {
          "height": 40,
          "marginBottom": 20,
          "width": 40,
        }
      }
    />
    <View>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          {
            "color": "#FFF",
            "fontSize": 20,
            "fontWeight": "bold",
            "width": "80%",
          }
        }
      >
        name
      </Text>
      <Text
        style={
          {
            "color": "#eef",
            "fontSize": 14,
            "marginBottom": 10,
            "marginTop": 5,
          }
        }
      >
        email
      </Text>
    </View>
  </View>
  <Text
    style={
      {
        "color": "#FFF",
        "fontSize": 16,
      }
    }
  >
    body
  </Text>
</View>
`;
